sdk/agent/agent_test.go

```diff
@@ -66,21 +107,25 @@ func TestAgent_openPaymentClose(t *testing.T) {
 			}
 			return 0, fmt.Errorf("unknown escrow account")
 		}),
-		balanceCollector: balanceCollectorFunc(func(accountID *keypair.FromAddress, asset state.Asset) (int64, error) {
+		BalanceCollector: balanceCollectorFunc(func(accountID *keypair.FromAddress, asset state.Asset) (int64, error) {
 			return 100_0000000, nil
 		}),
-		submitter: submitterFunc(func(tx *txnbuild.Transaction) error {
+		Submitter: submitterFunc(func(tx *txnbuild.Transaction) error {
 			localVars.submittedTx = tx
 			return nil
 		}),
-		streamer: streamerFunc(func(cursor string, accounts ...*keypair.FromAddress) (transactions <-chan StreamedTransaction, cancel func()) {
+		Streamer: streamerFunc(func(cursor string, accounts ...*keypair.FromAddress) (transactions <-chan StreamedTransaction, cancel func()) {
 			return localVars.transactionsStream, func() {}
 		}),
-		escrowAccountKey:    localEscrow.FromAddress(),
-		escrowAccountSigner: localSigner,
-		logWriter:           io.Discard,
-		events:              localEvents,
+		EscrowAccountKey:    localEscrow.FromAddress(),
+		EscrowAccountSigner: localSigner,
+		LogWriter:           io.Discard,
+		Events:              localEvents,
 	}
+	localConfig.Snapshoter = snapshoterFunc(func(a *Agent, s Snapshot) {
```

Comment:
Unfortunately we can't because the `localConfig` is referenced inside the `snapshotterFunc`, so it has to be set outside the definition of `localConfig`. This shouldn't have any functional/material impact though.